packages/components/src/site-thumbnail/use-mshots-img.tsx

```diff
@@ -25,18 +25,29 @@ export type MShotsOptions = {
 
 export const useMshotsImg = (
 	src: string,
-	options: MShotsOptions
+	options: MShotsOptions,
+	sizes: Array< { width: number; height: number } > = []
 ): {
-	src: string;
 	isLoading: boolean;
 	isError: boolean;
 	imgRef: React.MutableRefObject< HTMLImageElement | null >;
+	imgProps: Partial< React.ImgHTMLAttributes< HTMLImageElement > >;
 } => {
 	const [ retryCount, setRetryCount ] = useState( 0 );
-	const mshotUrl = useMemo(
-		() => mshotsUrl( src, options, retryCount ),
-		[ src, options, retryCount ]
-	);
+	const { mshotUrl, srcSet } = useMemo( () => {
+		const mshotUrl = mshotsUrl( src, options, retryCount );
+
+		// Calculate source image srcSet for all given sizes and retina 2x.
```

Comment:
Should we have some test coverage on this computation?

Also, some devices are actually 3x, so we might want to calculate those widths too.

I think https://github.com/ausi/respimagelint/ does some rounding to avoid too many `srcset` widths.